engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/ExecutionStore.scala

```diff
@@ -22,19 +21,23 @@ object ExecutionStore {
       case _ => None
     }
 
-    new ExecutionStore(keys.flatten.map(_ -> NotStarted).toMap)
+    new ExecutionStore(keys.flatten.map(_ -> NotStarted).toMap, keys.nonEmpty)
   }
 }
 
-case class ExecutionStore(private val store: Map[JobKey, ExecutionStatus]) {
+final case class ExecutionStore(private val statusStore: Map[JobKey, ExecutionStatus], hasNewRunnables: Boolean) {
```

Comment:
This was exactly why I balked at the `private val` FWIW. While I understand the argument of "it prevents us from possibly doing something stupid performance-wise" that seems to go against the grain of what a case class is inherently for, IMO it's fair to document "please don't do XYZ" and leave it at that.